client/txn.go

```diff
@@ -82,6 +82,17 @@ func newTxn(db DB, depth int) *Txn {
 	return txn
 }
 
+func newTxnFromProto(db DB, depth int, t proto.Transaction) *Txn {
+	txn := newTxn(db, depth)
+	txn.txn = t
+	return txn
+}
+
+// GetState returns the transaction protobuf.
+func (txn *Txn) GetState() *proto.Transaction {
```

Comment:
Perhaps make this return `proto.Transaction` instead of a pointer. I'm also wondering if perhaps `Txn.txn` should be exported. The `client` package will soon be entirely internal...